http-server/src/main/java/org/triplea/server/access/AuthenticatedUser.java

```diff
@@ -16,20 +16,33 @@
  * that throws an {@code UnsupportedOperationException} if accessed.
  */
 @Builder
+@EqualsAndHashCode
 public class AuthenticatedUser implements Principal {
   @Getter @Nullable private final Integer userId;
   @Getter @Nonnull private final String userRole;
-  @Getter @Nullable private final ApiKey apiKey;
+  @Getter @Nonnull private final ApiKey apiKey;
+  @Nullable private final String name;
 
+  @Nullable
   @Override
   public String getName() {
-    throw new UnsupportedOperationException("Name lookup is not done on authentication.");
+    Preconditions.checkState(
+        (name == null && userRole.equals(UserRole.HOST))
+            || (name != null && !userRole.equals(UserRole.HOST)),
+        "All user roles will have a name except the 'HOST' role (lobby watcher)");
+
+    return name;
   }
 
   public int getUserIdOrThrow() {
     Preconditions.checkState(
-        !userRole.equals(UserRole.ANONYMOUS), "Anonymous users always have a null user id");
-    return Optional.ofNullable(userId)
-        .orElseThrow(() -> new AssertionError("Error, expected to have a non-null user id"));
+        userId != null, "This method is called when we expect user id to not be null");
```

Comment:
Was a bit curious about this, looking at the javadocs:

> checkArgument(boolean expression) Ensures the truth of an expression involving one or more parameters to the calling method.

> checkNotNull(T reference) Ensures that an object reference passed as a parameter to the calling method is not null.

> checkState(boolean expression)
Ensures the truth of an expression involving the state of the calling instance, but not involving any parameters to the calling method.

It looks like in this case `checkState` is appropriate. I used that also because it really is the data state that is invalid. `null` values are actually fine, but in combination with other values, it indicates a bad data state in DB.

In short, it looks like when checking args, `checkArgument` or `checkNotNull` are the ones to use.